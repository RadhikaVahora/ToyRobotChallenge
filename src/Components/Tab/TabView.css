
.Tab1Header {
font-style: normal;
font-weight: normal;
font-size: 15px;
line-height: 20px;
letter-spacing: 0.535714px;
color: #000000;
width: 124px;
height: 4px;
position: absolute;
width: 124px;
height: 4px;
left:1px;
top: 20px;
right:495px;
bottom: 208px;
text-align: center;
}
.Tab2Header {
  font-style: normal;
  font-weight: normal;
  font-size: 15px;
  line-height: 20px;
  letter-spacing: 0.535714px;
  color: #000000;
  width: 124px;
  height: 4px;
  position: absolute;
  width: 124px;
  height: 4px;
  left: 130px;
  top: 20px;
  right:406px;
  bottom: 208px;
  text-align: center;
  }

.ActiveTab{
  text-decoration-line: underline;
  text-decoration-color: #8E00FF;
  text-decoration-style: solid;
  width: 124px;
  height: 4px;
}

.TextContent2{
  position: absolute;
width: 572px;
height: 80px;
left: 25px;
top: 153px;
right: 23px;
bottom: 16px;

font-family: 'Roboto';
font-style: normal;
font-weight: normal;
font-size: 15px;
line-height: 20px;
/* or 133% */
letter-spacing: 0.535714px;
color: #000000;
}

.TextContent1{
  position: absolute;
width: 572px;
height: 80px;
left: 25px;
top: 97px;
right: 57px;
bottom: 112px;

font-family: 'Roboto';
font-style: normal;
font-weight: normal;
font-size: 15px;
line-height: 20px;
/* or 133% */
letter-spacing: 0.535714px;
color: #000000;
}
.HeaderTitle{
  position: absolute;
width: 134px;
height: 32px;
left: 25px;
top: 57px;
right:461px;
bottom:160px;

font-family: 'Roboto';
font-style: normal;
font-weight: normal;
font-size: 24px;
line-height: 32px;
/* identical to box height, or 133% */
text-align: center;
letter-spacing: 0.857143px;
color: #000000;
}

.HeaderTitle2{
  position: absolute;
width: 261px;
height: 32px;
left: 25px;
top: 57px;
right:461px;
bottom:160px;

font-family: 'Roboto';
font-style: normal;
font-weight: normal;
font-size: 24px;
line-height: 32px;
/* identical to box height, or 133% */
text-align: center;
letter-spacing: 0.857143px;
color: #000000;
}

.ButtonStyle{
  position: absolute;
left: 486px;
right: 21px;
top: 188px;
bottom: 16px;
height: 36px;
width: 113px;
color: white;
background: #006DFF;
border: 2px solid #006DFF;
box-sizing: border-box;
border-radius: 8px;

}

.BoxLine{
  position: absolute;
width: 620px;
height: 1px;
left:1px;
top: 39px;
right:1px;
bottom: 208px;
border: 1px #E7E7E8;
border-style: solid;
}

.Tab1UnderLine{
  position: absolute;
width: 124px;
height: 4px;
left:1px;
top: 37px;
right:495px;
bottom: 208px;
background: #8E00FF;
}

.Tab2UnderLine{
  position: absolute;
width: 124px;
height: 4px;
left:125px;
top: 37px;
right:382px;
bottom: 208px;
background: #8E00FF;
}
