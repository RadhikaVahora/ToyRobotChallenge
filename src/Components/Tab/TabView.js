import React from 'react';
import './TabView.css';
import PropTypes from 'prop-types';


const TabView = (props) => {

    const onClickTabItem = () => {
        const{onTabClick,label} = props;
        onTabClick(label);
    }

    const{label,activeTab} = props;

    return(
        <div>
        <div className={label==='Tincidunt'?'Tab1Header': label==='Sodales'&& 'Tab2Header' }>
            <li onClick={onClickTabItem}>{label}</li>
        </div>
        <div className='BoxLine'/>
        <div>
            {activeTab === 'Tincidunt' &&(
                <div>
                    <div className='Tab1UnderLine'/>
                    <h1 className='HeaderTitle'>Donec vitae</h1>
                    <p className='TextContent1'>
                    Ut tincidunt tincidunt erat. Fusce convallis metus id felis luctus adipiscing. Vivamus quis mi. Nullam vel sem. 
                    Morbi ac felis.
                    </p>
                <p className='TextContent2'>
                    Maecenas vestibulum mollis diam.. Proin viverra, ligula sit amet ultrices semper, 
                    ligula arcu tristique sapien, a accumsan nisi mauris ac eros. Suspendisse potenti. 
                    In dui magna, posuere eget, vestibulum et, tempor auctor, justo.
                </p>
                </div>
            )}
            {activeTab === 'Sodales' &&(
                <div>
                    <div className='Tab2UnderLine'/>
                    <h1 className='HeaderTitle2'>Donec sodales sagittis</h1>
                    <p className='TextContent1'>
                    Maecenas vestibulum mollis diam.. Proin viverra, ligula sit amet ultrices semper, ligula arcu tristique sapien, 
                    a accumsan nisi mauris ac eros. Suspendisse potenti. In dui magna, posuere eget, vestibulum et, tempor auctor, justo.
                    </p>
                    <button className='ButtonStyle'>Continue</button>
                </div>
            )}
        </div>
        
        </div>
    );

}

TabView.propTypes = {
    label:PropTypes.string.isRequired,
    activeTab:PropTypes.string.isRequired,
}
export default TabView;