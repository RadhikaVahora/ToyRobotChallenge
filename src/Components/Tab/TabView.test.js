
import React from 'react';
import renderer from 'react-test-renderer';
import TabView from './TabView';

describe('TabView', () => {
    it('should render correctly for tab Tincidunt',() =>{
        const props={
            label: 'Tincidunt',
            onTabClick:jest.fn(),
            activeTab:'Tincidunt'
         }
        const wrapper = renderer.create(<TabView {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
    it('should render correctly for tab Tincidunt',() =>{
        const props={
            label: 'Sodales',
            onTabClick:jest.fn(),
            activeTab:'Sodales'
         }
        const wrapper = renderer.create(<TabView {...props} />);
        expect(wrapper).toMatchSnapshot();

    });
   
});