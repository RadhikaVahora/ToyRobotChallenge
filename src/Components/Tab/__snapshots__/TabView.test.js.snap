// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabView should render correctly for tab Tincidunt 1`] = `
<div>
  <div
    className="Tab1Header"
  >
    <li
      onClick={[Function]}
    >
      Tincidunt
    </li>
  </div>
  <div
    className="BoxLine"
  />
  <div>
    <div>
      <div
        className="Tab1UnderLine"
      />
      <h1
        className="HeaderTitle"
      >
        Donec vitae
      </h1>
      <p
        className="TextContent1"
      >
        Ut tincidunt tincidunt erat. Fusce convallis metus id felis luctus adipiscing. Vivamus quis mi. Nullam vel sem. Morbi ac felis.
      </p>
      <p
        className="TextContent2"
      >
        Maecenas vestibulum mollis diam.. Proin viverra, ligula sit amet ultrices semper, ligula arcu tristique sapien, a accumsan nisi mauris ac eros. Suspendisse potenti. In dui magna, posuere eget, vestibulum et, tempor auctor, justo.
      </p>
    </div>
  </div>
</div>
`;

exports[`TabView should render correctly for tab Tincidunt 2`] = `
<div>
  <div
    className="Tab2Header"
  >
    <li
      onClick={[Function]}
    >
      Sodales
    </li>
  </div>
  <div
    className="BoxLine"
  />
  <div>
    <div>
      <div
        className="Tab2UnderLine"
      />
      <h1
        className="HeaderTitle2"
      >
        Donec sodales sagittis
      </h1>
      <p
        className="TextContent1"
      >
        Maecenas vestibulum mollis diam.. Proin viverra, ligula sit amet ultrices semper, ligula arcu tristique sapien, a accumsan nisi mauris ac eros. Suspendisse potenti. In dui magna, posuere eget, vestibulum et, tempor auctor, justo.
      </p>
      <button
        className="ButtonStyle"
      >
        Continue
      </button>
    </div>
  </div>
</div>
`;
